### DISCUSSION TITLE
Newsletter

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#219] [General]
Hi Franky.
I would like to restrict the events listed in the newsletter according to the category of the event and the period concerned or the recurrence. Is this possible?

**Comment by liedekef** [comment]
#_MAILGROUPS can take an optional list of group ids, not related to events per se.
If that is not what you want, I don't understand what you're asking :-)  The newsletter functionality is just a generic automated task to inform people about new events, not related to mailing lists and also unaware of possible restrictions since it runs independantly on a daily basis.

**Reply by f14m07** [reply]
I wanted to know if I could restrict the list of events by specifying certain events and not others. For example, by using [eme_if] in the body, which works in some mails.

**Reply by liedekef** [reply]
You don't specify where/what. You just mentioned "newsletter", nothing else. Describe the setup, used shortcodes, config, etc ... first.

**Reply by f14m07** [reply]
As stated in the documentation, there is the header, the body and the footer. In the body I just have this model 
```<ul>
 	<li>#_LINKEDNAME avec #_CONTACTDISPNAME, le <strong>#_STARTDATE</strong>, <span style="color: #ff0000;">#_AVAILABLESEATS</span> place(s) encore disponible(s)</li>
</ul>`

**Reply by liedekef** [reply]
I'm assuming you're talking about: https://www.e-dynamics.be/wordpress/newsletter/
It can use all event placeholders (and thus also eme_if shortcodes as conditionals). However, if the result is empty, you need to make sure the whole resulting body is empty as otherwise EME will send a mail. Currently EME is not ready for that, this changeset should help:
https://github.com/liedekef/events-made-easy/commit/9c5272e1871e613db91f5826b43afcb980327803

**Reply by f14m07** [reply]
Thanks many. I will make some tests on my local site.

**Reply by f14m07** [reply]
It's perfect eme_if works very well in the template
